extensions/pkg/controller/heartbeat/add.go

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
```

Comment:
To be in line with the other controller packages, should we move the content of this file to `controller.go`?

Reply:
Named this as `add.go` because it does the same stuff as in e.g. https://github.com/gardener/gardener/blob/master/pkg/provider-local/controller/worker/add.go
but since there won't be any differing implementations it's part of the `extensions/pkg/controller/heartbeat` package directly. Anyway, I can move it to `controller.go` so that we reduce the number of files as it doesn't contain too much code.